Library/Homebrew/dev-cmd/prof.rb

```diff
@@ -17,7 +17,7 @@ def prof_args
   end
 
   def prof
-    prof_args.parse
+    prof_args.parse(ignore_invalid_options: true)
```

Comment:
Ideally this would still error if no arguments are given. Or maybe it should just profile `brew` without argument in that case? In that case, we can also remove the `min_named` above.

Reply:
Yeah I just realised that. That's why I proposed an alternative solution of `-- --prefix`, as that would turn `--prefix` into a named argument. All that would be needed is to then use `args.named_args` instead of `ARGV` a few lines down.

Whatever would make most sense to enter on a command line.

Reply:
"`--` as end-of-options marker" is pretty much Unix lore by now, so I think that makes more sense from a user POV.